components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/processor/DCRProcessor.java

```diff
@@ -52,6 +54,22 @@ public IdentityResponse.IdentityResponseBuilder process(IdentityRequest identity
         if (log.isDebugEnabled()) {
             log.debug("Request processing started by DCRProcessor.");
         }
+
+        boolean isIdentityRegisterEnabled =
+                IdentityUtil.isLegacyFeatureEnabled(DCRConstants.DCR_ID, DCRConstants.DCR_VERSION);
+
+        if (!isIdentityRegisterEnabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("Identity Register endpoint was deprecated. To enable the DCR API endpoint add the " +
+                        "following config to deployment.toml file. \n " +
+                        "[[legacy_feature]] \n" +
+                        "id = identity/register \n" +
+                        "enable = true");
+            }
+            String errorMessage = "/identity/register API was deprecated. Refer the WSO2 documentation for more " +
+                    "information (https://is.docs.wso2.com).";
```

Comment:
let's not send out WSO2-specific details in the error response to the client. This will expose the internal deployment information

Reply:
I followed the [same format](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java#:~:text=%22/identity/connect/register%20API) as the `identity/connect/register` for the error message.

Related PR: https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1447

To maintain the consistency it would be better to edit both error messages.

Reply:
Remove in the older class as well, and update this PR

Reply:
Suggested changes are applied.